_posts/2025-03-15-cryostat-400-release.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title: Cryostat 4.0.0 is Released!
+date: 2025-03-01
+synopsis: Cryostat 4.0.0 is here. Learn what's new in this release.
+author: Andrew Azores
+---
+
+#### Table of Contents
+* auto-gen TOC:
+{:toc}
+<hr>
+
+Hello everyone,
+
+It's time for another Cryostat release. This time, we are very proud to present Cryostat 4.0, a new major version with again some significant under-the-hood changes,
+and some exciting new OpenShift-specific features.
+
+### Feature Highlights
+1. **Quarkus and Hibernate JFR integrations** for framework-specific *Flight Recorder* events. If your target application uses Quarkus and contains the `quarkus-jfr` extension (along with `quarkus-rest`), or has `hibernate-jfr` on its classpath, then you can take advantage of the new *Preset Event Templates* that ship with Cryostat. These are open to community contributions for any other popular frameworks and libraries that provide JFR events. Simply start a recording using the relevant preset template and begin capturing these framework-specific events.
+2. **Automated Rules by JFR Event Type ID**: there is a new `jfrEventTypeIds(target): string[]` function available in the `matchExpression` Common Expression Language scope. Pass in the `target` object also in scope and this function returns a list of JFR event type IDs for that target JVM. This enables functionality such as that seen in the new `quarkus` and `hibernate` Automated Rules that ship with Cryostat 4.0, where the rules' expressions are satisfied by target JVMs that support the relevant JFR event types. This makes it even quicker and easier to get started capturing data from applications using these frameworks and libraries.
+3. **Diagnostics Card**: the Cryostat dashboard has a new *Diagnostics* card, which currently exposes one new piece of functionality: you can fire a request for the remote target JVM to run a Garbage Collection cycle. This can be useful when troubleshooting a misbehaving application, or when attempting to optimize the resource allocation for a container.
```

Comment:
Should we mention console is now upgraded/refined to Patternfly 5?

Reply:
I mentioned PF5 as a maintenance item further down:

https://github.com/cryostatio/cryostatio.github.io/pull/217/files#diff-ae235f1f555a166420f30932897c416f70c8a31e087c1e88530b1ecb95440446R47

Reply:
oh right, sorry no idea how I missed that! Thanks :D